runtime/src/main/java/io/quarkiverse/fx/FxApplication.java

```diff
@@ -1,12 +1,26 @@
 package io.quarkiverse.fx;
 
+import java.util.concurrent.CountDownLatch;
+
+import jakarta.enterprise.inject.Produces;
 import jakarta.enterprise.inject.spi.CDI;
 import jakarta.enterprise.util.AnnotationLiteral;
 
 import javafx.application.Application;
 import javafx.stage.Stage;
 
 public class FxApplication extends Application {
+    private static CountDownLatch started = new CountDownLatch(1);
```

Comment:
I would have made it non-static, and final.

Reply:
It has to be static or else it will not be accessible across the instances that are created. The javafx Application#launch will create an instance outside of CDI, and CDI will create one or more instances depending on how may injection sites there are for the CountDownLatch.

Reply:
Did you make `FxApplication` `@ApplicationScoped` ?

Reply:
Now FxApplication is `@ApplicationScoped`, all injected latch instances are the same

Reply:
Yes, but the instance created by the javafx.application.Application#launch method that calls FxApplication#start is not a CDI instance. It is a non-contextual instance that CDI does not manage. It is this CountDownLatch that controls the startup synchronization.

Reply:
See the latest update where I have separated the CDI and non-CDI instances. FxApplication remains a non-CDI bean that programmatically looks up the new StartupLatch singleton wrapper of a CountDownLatch and uses that to notifying any waiting threads. This keeps the lines of responsibility cleaner.